src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -294,11 +294,10 @@ public TagHelperContent EndTagHelperWritingScope()
 
                 foreach (var attribute in tagHelperOutput.Attributes)
                 {
-                    var value = HtmlEncoder.HtmlEncode(attribute.Value);
                     writer.Write(' ');
                     writer.Write(attribute.Key);
                     writer.Write("=\"");
-                    writer.Write(value);
+                    WriteTo(writer, HtmlEncoder, attribute.Value, escapeQuotes: true);
```

Comment:
- why is `HtmlString` a better location for this?  that class is used all over the place whereas this `WriteTo` method is called three times.
- this `WriteTo` method is `static` and so cannot reference an instance property of the page.